crates/chain-config/src/config/codec/parquet/encode.rs

```diff
@@ -0,0 +1,122 @@
+use std::{
+    io::Write,
+    marker::PhantomData,
+    sync::Arc,
+};
+
+use fuel_core_types::fuel_types::canonical;
+use itertools::Itertools;
+use parquet::{
+    basic::{
+        Compression,
+        Repetition,
+    },
+    file::{
+        properties::WriterProperties,
+        writer::SerializedFileWriter,
+    },
+};
+
+use parquet::{
+    data_type::ByteArrayType,
+    schema::types::Type,
+};
+
+pub struct Encoder<W: Write + Send, T, E> {
+    writer: SerializedFileWriter<W>,
+    _type: PhantomData<T>,
+    _encoding: PhantomData<E>,
+}
+
+pub type PostcardEncoder<T> = Encoder<std::fs::File, T, PostcardEncode>;
+
+impl<W: Write + Send, T, E> Encoder<W, T, E>
+where
+    E: Encode<T>,
+{
+    pub fn new(writer: W, compression: Compression) -> anyhow::Result<Self> {
+        let writer = SerializedFileWriter::new(
+            writer,
+            Arc::new(Self::single_element_schema()),
+            Arc::new(
+                WriterProperties::builder()
+                    .set_compression(compression)
+                    .build(),
+            ),
+        )?;
+
+        Ok(Self {
+            writer,
+            _type: PhantomData,
+            _encoding: PhantomData,
+        })
+    }
+
+    fn single_element_schema() -> Type {
+        let data =
+            Type::primitive_type_builder("data", ::parquet::basic::Type::BYTE_ARRAY)
+                .with_repetition(Repetition::REQUIRED)
+                .build()
+                .expect("This is a valid schema");
+
+        Type::group_type_builder("unimportant")
+            .with_fields(vec![Arc::new(data)])
+            .build()
+            .expect("This is a valid schema")
+    }
+}
+
+impl<W, T, E> Encoder<W, T, E>
+where
+    W: Write + Send,
+    E: Encode<T>,
+{
+    pub fn write(&mut self, elements: Vec<T>) -> anyhow::Result<()> {
+        let mut group = self.writer.next_row_group()?;
+        let mut column = group
+            .next_column()?
+            .ok_or_else(|| anyhow::anyhow!("Missing column. Check the schema!"))?;
+
+        let values: Vec<_> = elements
```

Comment:
This might not play well with parquet. AFAIK it's not straightforward to go from a row which is supposed to be a single element to treating it like it's a collection. 
I'll set this to resolved to move forward with the feature unless you think it's worth it to investigate further.